### ISSUE TITLE
Long puzzle solving time

### ISSUE THREAD

**Issue by lenni-io** [FriendlyCaptcha/friendly-challenge#156]
Hey FriendlyCaptcha Team,

we are currently using your Captcha on multiple websites ( Script Embed of the Widget ). We never really had a problem with it. But right now we are the Javascript API for a specific use case and the puzzle solving take very long time. Something around 1-2 minutes.

What could be possible reasons which could cause such a slow puzzle solving?

Please let me know, if you have any further questions.

**Comment by gzuidhof** [comment]
Hey lenni-io,

Could you create a support ticket for this with a few more details? (E.g. the sitekey, perhaps a link to the webpage, and if you know perhaps the affected browsers?)

Sometimes this can be caused by the content-security-policy (CSP) headers being set wrong, which prevents some browsers (such as Safari) from being allowed to use WebAssembly. It would then have to use the fallback Javascript code, which can be a lot slower.

So in short: let's figure it out together in e-mail :)
